src/consensus.rs

```diff
@@ -231,23 +231,20 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                         self.persistent_log.entry(LogIndex(prev_log_index)).unwrap().0
                     };
 
-                let mut message = MallocMessageBuilder::new_default();
-                {
-                    let mut request = message.init_root::<message::Builder>()
-                                             .init_append_entries_request();
-                    request.set_term(self.current_term().into());
-
-                    request.set_prev_log_index(prev_log_index);
-                    request.set_prev_log_term(prev_log_term.into());
-                    request.set_leader_commit(self.commit_index.into());
-
-                    let mut entries = request.init_entries((until_index - from_index) as u32);
-                    for (n, index) in (from_index..until_index).enumerate() {
-                        entries.set(n as u32, self.persistent_log.entry(LogIndex::from(index)).unwrap().1);
-                    }
+                let mut entries = vec![];
+                for index in (from_index .. until_index) {
+                    entries.push(self.persistent_log.entry(LogIndex::from(index)).unwrap())
```

Comment:
not directly related to this PR, but in the normal case, you're going to read multiple entries. I think it would make sense if the persistent log interface reflected that by taking the format
`entries(lo, hi)` where `hi` is exclusive. So you'd get
`let entries = self.persistent_log.entry(index, until_index)` and the persistent log would be able to help you out a little bit performance wise (on top of the small gain here).